python/core/qgsfields.sip

```diff
@@ -1,253 +1,332 @@
+.. note::
+
+   QgsFields objects are implicitly shared.
+%End
+
 %TypeHeaderCode
-#include <qgsfield.h>
+#include "qgsfields.h"
 %End
   public:
 
     enum FieldOrigin
     {
-      OriginUnknown,   //!< it has not been specified where the field comes from
-      OriginProvider,  //!< field comes from the underlying data provider of the vector layer  (originIndex = index in provider's fields)
-      OriginJoin,      //!< field comes from a joined layer   (originIndex / 1000 = index of the join, originIndex % 1000 = index within the join)
-      OriginEdit,      //!< field has been temporarily added in editing mode (originIndex = index in the list of added attributes)
-      OriginExpression //!< field is calculated from an expression
+      OriginUnknown,
+      OriginProvider,
+      OriginJoin,
+      OriginEdit,
+      OriginExpression
     };
 
-    /** Constructor for an empty field container
-     */
+
     QgsFields();
+%Docstring
+ Constructor for an empty field container
+%End
+
+    QgsFields( const QgsFields &other );
+%Docstring
+ Copy constructor
+%End
 
-    /** Copy constructor
-     */
-    QgsFields( const QgsFields& other );
 
     virtual ~QgsFields();
 
-    //! Remove all fields
     void clear();
-    //! Append a field. The field must have unique name, otherwise it is rejected (returns false)
-    bool append( const QgsField& field, FieldOrigin origin = OriginProvider, int originIndex = -1 );
-    //! Append an expression field. The field must have unique name, otherwise it is rejected (returns false)
-    bool appendExpressionField( const QgsField& field, int originIndex );
-    //! Remove a field with the given index
+%Docstring
+Remove all fields
+%End
+
+    bool append( const QgsField &field, FieldOrigin origin = OriginProvider, int originIndex = -1 );
+%Docstring
+Append a field. The field must have unique name, otherwise it is rejected (returns false)
+ :rtype: bool
+%End
+
+    bool appendExpressionField( const QgsField &field, int originIndex );
+%Docstring
+Append an expression field. The field must have unique name, otherwise it is rejected (returns false)
+ :rtype: bool
+%End
+
     void remove( int fieldIdx );
+%Docstring
+Remove a field with the given index
+%End
 %MethodCode
-  if ( a0 < 0 || a0 >= sipCpp->count() )
-  {
-    PyErr_SetString(PyExc_KeyError, QByteArray::number(a0));
-    sipIsErr = 1;
-  }
-  else
-  {
-    sipCpp->remove( a0 );
-  }
+    if ( a0 < 0 || a0 >= sipCpp->count() )
+    {
+      PyErr_SetString( PyExc_KeyError, QByteArray::number( a0 ) );
+      sipIsErr = 1;
+    }
+    else
+    {
+      sipCpp->remove( a0 );
+    }
 %End
 
-    //! Extend with fields from another QgsFields container
-    void extend( const QgsFields& other );
+    void extend( const QgsFields &other );
+%Docstring
+Extend with fields from another QgsFields container
+%End
 
-    //! Check whether the container is empty
     bool isEmpty() const;
-    //! Return number of items
-    int count() const;
```

Comment:
Sure, but in that case it'll need adding to the API break docs, and checking over the python plugin source to make sure it's not used there.